feature/rds/auth/go.mod

```diff
@@ -0,0 +1,7 @@
+module github.com/aws/aws-sdk-go-v2/feature/rds/auth
+
+go 1.15
```

Comment:
can we make this util have min version go 1.16?

Reply:
I don't think we want to manage user accessible modules in that manner. Otherwise it's a pretty confusing experience on communicating what versions of the Go compiler we support. For modules that are not internal I think it's best we are consistent with what the minimum version is across the board. Obvious exceptions to that statement when we start major versioning components and "deprecating" old versions etc.